### PR TITLE
Auto-set template name based on action

### PR DESCRIPTION
**Pull request by jadb**

### CODE REVIEW

File: src/Mailer/Mailer.php [hunk 1]
```diff
@@ -226,6 +226,10 @@ public function send($action, $args = [], $headers = [])
 
         call_user_func_array([$this, $action], $args);
 
+        if (empty($this->template)) {
```
Comment (markstory) [inline src/Mailer/Mailer.php]: Do we ever thing a joker will use '0.ctp' ?

Comment (bcrowe) [inline src/Mailer/Mailer.php]: Ain't no joke. The struggle is real.

Comment (jadb) [inline src/Mailer/Mailer.php]: Should I replace by `is_null()`?

Comment (ADmad) [inline src/Mailer/Mailer.php]: We use `=== null`.

Comment (jadb) [inline src/Mailer/Mailer.php]: Keep forgetting, will do that.

**Comment by ADmad** [comment]
@jadb Tests failing

**Comment by markstory** [comment]
ping.

**Comment by jadb** [comment]
Looks like CS errors unrelated to this PR.

**Comment by markstory** [comment]
Should this have new tests added?

**Comment by markstory** [comment]
ping

**Comment by jadb** [comment]
I'll do it, sorry missed it. Need to start clearing the stuff I assign to myself for a change, been really busy.

**Comment by jadb** [comment]
Sorry for the delay. I added the tests and amended the other commit.

**Comment by ADmad** [comment]
With this change setting custom template using `$this->_email->template('custom')` in a mailer method no longer works. It still ends up using the method name as template.

**Comment by markstory** [comment]
@ADmad Seems like something that should be fixed.

**Comment by ADmad** [comment]
@markstory I have opened the new issue #7278 explaining the problems with mailer.

**Pull request merged**